otter/integration/tests/test_convergence.py

```diff
@@ -989,6 +990,101 @@ def test_recover_from_identity_auth_failures(self):
             max_entities=10)
         yield self.helper.start_group_and_wait(group, rcs, desired=5)
 
+    @inlineCallbacks
+    def test_error_reasons_are_updated(self):
+        """
+        Error reasons appear if the group goes into an ERROR state, disappear
+        when the group fixes itself, and reappears (but with a new error
+        reason) if the group goes into ERROR again.
+
+        1. Create group with an invalid key name, because autoscale does not
+           verify the key and hence won't just 400 the create group request.
+        2. Assert group goes into error state with the reason being invalid
+           Server config.
+        3. Delete the key, and converge.
+        4. Assert group goes into active state.
+        5. Change the launch config to have an invalid load balancer.
+        6. Scale up.
+        7. Assert group goes into error state with the reason being an invalid
+           CLB.
+        """
+        group, server_name_prefix = self.helper.create_group(
+            min_entities=1, max_entities=10,
+            server_name_prefix="no-such-key",
+            key_name="invalid_key_name"
+        )
+        launch_config = deepcopy(group.group_config['launchConfiguration'])
+
+        if not not_mimic():
+            # if this is mimic, we have to make server creation fail.
+            # in production, it's actually an invalid key and hence would
+            # naturally fail.
```

Comment:
Mimic doesn't actually implement glance or the key store or anything.  It has some preset invalid images that you can use to fail the server, but it doesn't  have a set of preset keys.  We should probably add that, yeah.